runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSourceWrapper.java

```diff
@@ -67,40 +78,51 @@ public String getName() {
   @Override
   public void run(SourceContext<WindowedValue<T>> ctx) throws Exception {
```

Comment:
is it possible to inject the `PipelineOptions` in the `SourceContext`? That way they would be provided in the args to the `run` method and you would not need to manually serialize/deserialize this class.